R/bioicons.R

```diff
@@ -0,0 +1,85 @@
+#' @rdname bioicons
+#' @export
+download_bioicons <- function() {
+  icon_df <-
+    jsonlite::read_json(
+      "https://raw.githubusercontent.com/duerrsimon/bioicons/main/static/icons/icons.json",
+      simplifyVector = TRUE
+    )
+  icon_df$author <- gsub(" ", "_", icon_df$author)
+
+  url <- "https://github.com/duerrsimon/bioicons/archive/refs/heads/main.zip"
+
+  meta <-
+    jsonlite::read_json(
+      glue::glue("https://raw.githubusercontent.com/duerrsimon/bioicons/main/package.json")
+    )
+
+  install_icon_zip(
+    "bioicons", url, c("static", "icons"),
+    meta = list(
+      name = "Bioicons",
+      version = meta$version,
+      licence = c("BSD", "CC-0", "CC-BY-SA", "MIT")
+    )
+  )
+  saveRDS(
+    icon_df,
+    glue::glue("{rappdirs::user_data_dir('rpkg_icon/bioicons')}/icons.rds")
```

Comment:
I've changed this to use the `icon_path()` function. It allows users to specify where they want the icons stored, and also resolves issues with `/` not always being the appropriate file separator.